docs/40-categorized-add-block-dialog.md

```diff
@@ -0,0 +1,57 @@
+# Block selection dialog with categories
+
+![select-a-block.png](img/select-a-block.png)
+
+## Configuration
+
+To enable and configure this dialog (replacing the default dropdown and button) you have to configure it according to the following example:
+
+```
+sonata_helpers:
+  compose_container:
+    enabled: true
+    categories:
+      '':  "block.category.default"
+      foo: "Foo"
+      bar: "Bar"
+    block_config:
+      'App\Block\BannerBlockService':
+        categories:
+          - foo
+      'App\Block\SimpleTextBlockService':
+        categories:
+          - bar
+      'App\Block\SkillsBlockService':
+        categories:
+          - foo
+          - bar
+```
+
+* `enabled`: Set to `false` to completely disable the feature. (Or do not add this configuration at all.)
+* `categories`: Key-label-list of all available categories. The empty key is the default category, assigned to all blocks without specified categories.
+* `block_config`: Assign one or more categories to each Block class.
+
+
+## Installation
+
+This feature requires jQuery 2 or 3.
+
+You need to make the following files available to the web frontend:
+* `src/Resources/public/js/MutationObserver.js`
+* `src/Resources/public/css/blockSelectModal.scss`
+
+
+If for example you use Webpack-Encore, edit your `webpack.config.js`:
+
+```
+Encore.addStyleEntry('...', [
+  ...
+  path.resolve('/vendor/ekino/sonata/src/Resources/public/css/blockSelectModal.scss')
+]);
+...
+Encore.addEntry('...', [
+  ...
+  path.resolve('/vendor/ekino/sonata/src/Resources/public/js/MutationObserver.js')
+]);
+```
+
```

Comment:
we need to add something about the compiler pass here. We should explain that by enabling the feature, the PageAdminController will now be the one from this bundle and that it override `composeContainerShowAction` method from sonata page bundle one to make the feature available.

We could also add a word about that if you want this feature but you also need to override the page admin controller, then you should extend this PageAdminController